### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by simonsmith**
@giuseppeg I've simplified some of the code (not sure why I needed to return a function from `lintImportedFiles`) and adjusted the tests to be a bit more readable. Hoping this makes more sense and easier to build upon as we go forward.

I realised they were really hard to follow when I tried to read it again, my bad.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -179,37 +180,37 @@ describe('suitcss', function() {
...
           root: 'test/fixtures',
           beforeLint: beforeLintStub
         }).catch(done);
 
-        expect(bemLintStub.calledOnce).to.be.ok;
-        expect(beforeLintStub.calledOnce).to.be.ok;
-        expect(beforeLintStub.calledBefore(bemLintStub)).to.be.ok;
+        expect(lintImportedFilesStub).to.be.called.once;
+        expect(beforeLintStub).to.be.called.once;
+        expect(beforeLintStub).to.have.been.calledBefore(lintImportedFilesStub);
```
Comment (giuseppeg) [inline test/test.js]: This assertion alone should cover all the cases right?
It implicitly means that `beforeLintStub` and `lintImportedFilesStub` were called.

If I am right you can remove the two lines above

**Comment by giuseppeg** [comment]
@simonsmith looks good to me, feel free to merge it whenever you want.

File: test/test.js (hunk 1 above)
Comment (simonsmith) [inline test/test.js]: Not sure. It could be called before but still get called more than once. I'll check before I merge.

Comment (simonsmith) [inline test/test.js]: Okay, I think it's keeping as is, because from checking just now, `calledBefore` doesn't care about `callCount` so they both have value.
